corporate/lib/stripe.py

```diff
@@ -1249,24 +1312,35 @@ def setup_upgrade_payment_intent_and_charge(
     ) -> str:
         customer = self.update_or_create_stripe_customer()
         assert customer is not None  # for mypy
-        discount_for_plan = customer.get_discount_for_plan_tier(plan_tier)
-        price_per_license = get_price_per_license(plan_tier, billing_schedule, discount_for_plan)
+        fixed_price_plan = None
+        if plan_tier == customer.required_plan_tier:
+            fixed_price_plan = CustomerPlan.objects.filter(
+                customer=customer, fixed_price__isnull=False, status=CustomerPlan.NEVER_STARTED
+            ).last()
```

Comment:
Should this be written as `if customer.required_plan_tier is not None: assert plan_tier == customer.required_plan_tier`? I think that might make it more clear that we're expecting code prior to this to enforce `customer.required_plan_tier`.